File: 2206-divide-array-into-equal-pairs/2206-divide-array-into-equal-pairs.java
class Solution {
    public boolean divideArray(int[] nums) {
         Map<Integer,Integer> resss= new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            Integer j=resss.getOrDefault(nums[i],-1);
            resss.put(nums[i],(j==-1?1:j+1) );
        }
        for (Map.Entry<Integer, Integer> entry : resss.entrySet()) { 
            if(entry.getValue()%2==1)return false;
        }

        return true;
    }
}

File: 2206-divide-array-into-equal-pairs/2206-divide-array-into-equal-pairs.py
class Solution(object):
    def divideArray(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        nums.sort()
        db = len(nums) // 2

        for i in range(0, len(nums), 2):
            if nums[i:i + 2][0]!=nums[i:i + 2][1]:
                return  False
        return True
